core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -649,24 +650,77 @@ String getFeatureVariableValueForType(@Nonnull String featureKey,
         String variableValue = variable.getDefaultValue();
         Map<String, ?> copiedAttributes = copyAttributes(attributes);
         FeatureDecision featureDecision = decisionService.getVariationForFeature(featureFlag, userId, copiedAttributes);
-        if (featureDecision.variation != null) {
+        Boolean featureEnabled = false;
+        if (featureDecision.variation != null && featureDecision.variation.getFeatureEnabled()) {
             FeatureVariableUsageInstance featureVariableUsageInstance =
                 featureDecision.variation.getVariableIdToFeatureVariableUsageInstanceMap().get(variable.getId());
             if (featureVariableUsageInstance != null) {
                 variableValue = featureVariableUsageInstance.getValue();
             } else {
                 variableValue = variable.getDefaultValue();
             }
+            featureEnabled = featureDecision.variation.getFeatureEnabled();
         } else {
             logger.info("User \"{}\" was not bucketed into any variation for feature flag \"{}\". " +
                     "The default value \"{}\" for \"{}\" is being returned.",
                 userId, featureKey, variableValue, variableKey
             );
         }
 
+        Object convertedValue = convertStringToType(variableValue, variableType);
+        Map<String, Object> decisionInfo = new HashMap<>();
+        decisionInfo.put(DecisionInfoEnums.FeatureVariableDecisionInfo.FEATURE_KEY.toString(), featureKey);
```

Comment:
Not a valid use of Enums. What you're doing here just requires constants or can even be a builder.
Via constants this would reduce to:
```
decisionInfo.put(DECISION_FEATURE_KEY, featureKey);
decisionInfo.put(DECISION_FEATURE_ENABLED, featureEnabled);
...
```
Via a builder it would be:
```
Map<String, Object> decisionInfo = DecisionInfoMap.builder()
    .withFeatureKey(featureKey)
    .withFeatureEnabled(featureEnabled)
    ...
    .build();
```

I personally like the builder approach since within the builder you'd abstract some of the conditional logic below.
Actually, in the context of the notification listener the entire thing can benefit from the builder pattern which actually transforms into a fluent pattern if you in corporate the notification itself.

```
DecisionNotification.builder()
    .withUserId(userId)
    .withAttributes(attributes)
    .withFeatureKey(featureKey)
    .withFeatureEnabled(featureEnabled)
    ...
    .send();
```
A much cleaner interface :)